crates/sargon-uniffi/src/profile/mfa/security_structures/selected_factor_sources_status.rs

```diff
@@ -0,0 +1,29 @@
+use crate::prelude::*;
+use sargon::SelectedFactorSourcesForRoleStatus as InternalSelectedFactorSourcesForRoleStatus;
+
+/// Represents the status of selected factor sources for a specific role in the Security Shield building process.
+/// Primarily used for UI logic representation in host applications.
+#[derive(
+    Clone, Copy, Debug, PartialEq, Eq, InternalConversion, uniffi::Enum,
+)]
+pub enum SelectedFactorSourcesForRoleStatus {
+    /// The selected factor sources are optimal for the specified role
+    /// in the Security Shield building process.
+    Optimal,
+
+    /// The selected factor sources are suboptimal for the specified role
+    /// in the Security Shield building process.
+    ///
+    /// Note: Typically used in hosts as a warning message.
+    Suboptimal,
+
+    /// The selected factor sources are insufficient for the specified role
+    /// in the Security Shield building process.
+    Insufficient,
+
+    /// The selected factor sources form an invalid combination for the specified role
+    /// in the Security Shield building process.
+    ///
+    /// Example: A Password factor source cannot be used alone for the Primary role.
```

Comment:
This example seems to fall more under the `Insufficient` category

Reply:
From the host’s perspective, this should fall under the `Invalid` category and display a generic error message. If the Primary role has `threshold` set to `1`, then adding a Password factor source satisfies the threshold condition, but it’s still invalid. It could fall under the `Insufficient` category if we have a dynamic threshold that updates based on the available factor sources. Do we?

Reply:
Why is threshold important in the context of selecting the factors before going in the Access setup build screen?

Also,[ if there are no roles selected the status is Insufficient](https://github.com/radixdlt/sargon/blob/a37ad774dd6fafcaaa2ee6800cf1b92101630fe0/crates/sargon/src/profile/mfa/security_structures/security_shield_builder.rs#L455), but if you have password it is considered Invalid, when IMO it is a case of insufficient - you need one more factor.

Reply:
This is a very UI specific type and we have in the UI 3 error/warning states:
1. User need to select at least 1 factor -> `Insufficient`
2. It is recommended to select at least 2 factors -> `Suboptimal`
3. Any other invalid combinations for a specific role (Access) -> `Invalid`

Based on that, this case falls under `Invalid`, even if it doesn't sound correct

Reply:
I am confused here - what is any other combinations? Shouldn't we display only the factors that are allowed to be selected in treshold Access role?
So then the only "Invalid" scenario is that the password cannot be alone, some other might be added in the future sure.

> 1. User need to select at least 1 factor -> Insufficient
> 2. It is recommended to select at least 2 factors -> Suboptimal
> 3. Any other invalid combinations for a specific role (Access) -> Invalid

So the Wallet will display red for 1 and 3?